### PR TITLE
placeholder is now an nsurl

### PR DESCRIPTION
**Pull request by derrickstaten**
@shortstuffsushi this is needed for the new Facebook app. without this Facebook is no longer an option in share sheets

### CODE REVIEW

**Comment by derrickstaten** [comment]
Also, we need to push out an update to Cocoapods when we merge this, it's needed for at least one partner and should probably be there for all of them.

**Comment by shortstuffsushi** [comment]
The code looks OK here, but I haven't had a chance to try on a physical device. If you have, and you've tried the various core offerings (Twitter, Facebook, Mail, etc), then you can go ahead and merge :+1: 

As far as tagging, you can either do that when you merge, or I can do it later on today. Keep in mind that there have been a number of updates since the last tag when you're doing release notes. All in all, though, I think they're pretty minor changes, so 0.6.2 would still be an appropriate tag.

**Comment by derrickstaten** [comment]
I'll test across everything I have on my phone now. I'll let you do the actual push since I think you're probably faster at it and know what else belongs in the release notes

**Comment by derrickstaten** [comment]
@shortstuffsushi actually you should probably revisit this. currently we set the placeholder item at an NSString. Then in the item method, we check whether the placeholder item is a string. If it is, we know that we still need to generate a short URL. Perhaps there's a better way to do this?

I tried a few variations but decided that you're probably better suited to do this. Also happy to talk about it and do it myself after a short back-and-forth

**Comment by derrickstaten** [comment]
@shortstuffsushi this is ready now. please test extensively before incorporating

**Comment by shortstuffsushi** [comment]
I think this looks good, and haven't been able to find any issue while testing. I think it would be good if @scotthasbrouck took a look though, since he wrote this initially.

File: Branch-SDK/Branch-SDK/BranchActivityItemProvider.m [hunk 1]
```diff
@@ -37,21 +37,18 @@ - (id)initWithParams:(NSDictionary *)params
...
-            self.branchURL = [[Branch getInstance] getShortURLWithParams:self.params andTags:self.tags andChannel:channel andFeature:self.feature andStage:self.stage andAlias:self.alias];
-        }
-
-        return [NSURL URLWithString:self.branchURL];
+    NSString *channel = [BranchActivityItemProvider humanReadableChannelWithActivityType:self.activityType];
+    
+    // Because Facebook immediately scrapes URLs, we add an additional parameter to the existing list, telling the backend to ignore the first click
+    BOOL ignoreFirstClick = [channel isEqualToString:@"facebook"];
+    if (ignoreFirstClick) {
+        self.branchURL = [[Branch getInstance] getShortURLWithParams:self.params andTags:self.tags andChannel:channel andFeature:self.feature andStage:self.stage andAlias:self.alias ignoreUAString:self.userAgentString];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BranchActivityItemProvider.m]: Is there a reason to even have `self.branchURL` as a property?

Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/BranchActivityItemProvider.m]: i'm not actually sure. i was wondering this myself when we first added it

Comment (scotthasbrouck) [inline Branch-SDK/Branch-SDK/BranchActivityItemProvider.m]: Totally correct, `self.branchURL` is not needed. Pushing up a commit that removes it now. The rest of this looks solid.

**Comment by scotthasbrouck** [comment]
Ok, sorry for the confusion. That commit removes the `if` check on placeholder, and returns the placeholder to being an NSURL, and is what I meant to commit.

**Pull request merged**